client/src/components/forms/CommentEditor.vue

```diff
@@ -1,205 +1,216 @@
-      >
-        <q-tooltip class="bg-primary">Unset link</q-tooltip>
-        <q-icon name="link_off"></q-icon>
-      </q-btn>
+      <comment-editor-button
+        v-for="(button, index) in commentEditorButtons"
+        :key="index"
+        v-bind="button"
+      />
     </q-btn-group>
     <div class="editor">
-      <editor-content
-        :editor="editor"
-        style="
-          background: #dddddd;
-          min-height: 200px !important;
-          border-radius: 5px;
-        "
-        class="q-pa-xs"
-      />
+      <editor-content :editor="editor" />
     </div>
     <div class="q-pa-md q-gutter-y-sm column">
-      <q-toggle v-model="relevance" label="Relevance" />
-      <q-toggle v-model="accessibility" label="Accessibility" />
-      <q-toggle v-model="coherence" label="Coherence" />
-      <q-toggle v-model="scholarlydialogue" label="Scholarly Dialogue" />
+      <q-toggle
+        v-for="criteria in styleCriteria"
+        :key="criteria.id"
+        v-model="criteria.selected"
+        :data-ref="criteria.refAttr"
+        :label="criteria.label"
+      />
     </div>
-    <q-btn color="primary">Submit</q-btn>
-  </div>
+    <q-btn data-ref="submit" color="primary" @click="submitHandler()">{{
+      $t("guiElements.form.submit")
+    }}</q-btn>
+  </q-card>
 </template>
 
-<script>
+<script setup>
+import { ref, computed } from "vue"
 import { useEditor, EditorContent } from "@tiptap/vue-3"
+import { useQuasar } from "quasar"
 import StarterKit from "@tiptap/starter-kit"
-import Bold from "@tiptap/extension-bold"
-import Italic from "@tiptap/extension-italic"
-import BulletList from "@tiptap/extension-bullet-list"
-import OrderedList from "@tiptap/extension-ordered-list"
 import Link from "@tiptap/extension-link"
 import Placeholder from "@tiptap/extension-placeholder"
-import { ref } from "vue"
+import CommentEditorButton from "../atoms/CommentEditorButton.vue"
+import BypassStyleCriteriaDialogVue from "../dialogs/BypassStyleCriteriaDialog.vue"
+import { useI18n } from "vue-i18n"
+
+const { dialog } = useQuasar()
+function dirtyDialog() {
+  return dialog({
+    component: BypassStyleCriteriaDialogVue,
+  })
+}
+
+const { t } = useI18n()
+const editor = useEditor({
+  injectCSS: true,
+  extensions: [
+    StarterKit.configure({
+      blockquote: false,
+      codeblock: false,
+      hardbreak: false,
+      heading: false,
+      horizontalrule: false,
+      strike: false,
+    }),
+    Link.configure({
+      openOnClick: false,
+    }),
+    Placeholder.configure({
+      placeholder: t("submissions.comment.placeholder"),
+    }),
+  ],
+})
 
-export default {
-  components: {
-    EditorContent,
+const commentEditorButtons = ref([
+  {
+    ariaLabel: "guiElements.button.bold.ariaLabel",
+    isActive: computed(() => editor.value.isActive("bold")),
+    clickHandler: () => editor.value.chain().focus().toggleBold().run(),
+    tooltipText: "guiElements.button.bold.tooltipText",
```

Comment:
```suggestion
    tooltipText: t("guiElements.button.bold.tooltipText"),
```
related to my earlier comment

Reply:
This has been refactored.